core/src/main/scala/pickling/Pickler.scala

```diff
@@ -26,6 +27,7 @@ trait SPickler[T] {
 
 object SPickler {
   def generate[T]: SPickler[T] = macro Compat.PicklerMacros_impl[T]
+  implicit def spuToSp[T: SPicklerUnpickler]: SPickler[T] = implicitly[SPicklerUnpickler[T]].pickler
```

Comment:
With the `extends` approach, picklers needs to be born with unpickler, so that'll affect the pickler generating code, which I don't think it's worth it. Also I don't know what happens when you have `SPicklerUnpickler[Apple]` and `SPickler[Apple]` in scope. With `extends` approach you'd get collision?